daffodil-runtime1-unparser/src/main/scala/edu/illinois/ncsa/daffodil/processors/unparsers/IBM4690PackedDecimalUnparsers.scala

```diff
@@ -54,7 +54,21 @@ final class IBM4690PackedIntegerMinLengthInBytesUnparser(
   extends IBM4690PackedIntegerBaseUnparser(e) {
 
   override def getBitLength(state: ParseOrUnparseState): Int = {
-    val len = state.currentNode.get.asSimple.dataValue.asInstanceOf[Array[Byte]].length * 8
+    val num = state.currentNode.get.asSimple.dataValue.asInstanceOf[JBigInteger]
+    // If negative account for sign nibble
```

Comment:
If I understand this correctly, IBM4690 packed numbers only have sign nibbles when they are negative. If positive there is no sign nibble. Is that right? If so, please add a comment to the code stating this, as it is quite unintuitive, and otherwise someone might "fix" this code to be symmetric thinking it is broken, etc.